### PR TITLE
Update instructions for Challenge 1

### PR DESCRIPTION
**Pull request by RemakingEden**
Updating the instructions based on this issue. https://github.com/commjoen/wrongsecrets/issues/255

### CODE REVIEW

File: src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge4.java [hunk 1]
```diff
@@ -31,7 +31,8 @@ public Spoiler spoiler() {
 
     @Override
     public boolean answerCorrect(String answer) {
-        return argBasedPassword.equals(answer);
+        return argBasedPassword.equals(answer) 
+        || argBasedPassword.equals("'" + answer + "'");
```
Comment (commjoen) [inline src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge4.java]: looks like this is an older MR coming back. This one is already fixed. Can you remove the Challenge4 related changes? as these are already merged to master

File: src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge4Test.java [hunk 2]
```diff
@@ -26,6 +26,14 @@ void spoilerShouldRevealAnswer() {
     void rightAnswerShouldSolveChallenge() {
         var challenge = new Challenge4(scoreCard, "test");
 
+        Assertions.assertThat(challenge.solved("test")).isTrue();
```
Comment (commjoen) [inline src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge4Test.java]: looks like this is an older MR coming back. This one is already fixed. Can you remove the Challenge4 related changes please? As these are already merged to master

File: src/main/resources/explanations/challenge1.adoc [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 When people write a Proof of Concept, they often start with hardcoded secrets, such as a `password` in code. What if we forget to remove these hardcoded secrets?
 
-Can you spot the secret we are looking for in code? What about looking for it at the container?
+Can you spot the secret we are looking for in code? What about looking for it in the container?
 
-You can spot the secret easily by using https://github.com/trufflesecurity/truffleHog[*Trufflehog*] and https://github.com/awslabs/git-secrets[*Git-secrets*]. Just dive into the code!
+Sometimes the simpler tools are the most effective. Try cloning the repo and use https://man7.org/linux/man-pages/man1/grep.1.html[*grep*] to see what you find. 
+
+You can also find the answer using https://github.com/awslabs/git-secrets[*Git-secrets*] once you have added this rule `git secrets --add 'password\s*=\s*.+'`. Just dive into the code!
```
Comment (commjoen) [inline src/main/resources/explanations/challenge1.adoc:9]: Can you please update challenge1_hint.adoc in which git-secrets is setup as a third step? that way everyone can use it :)

**Review by commjoen — CHANGES_REQUESTED** [review]
Awesome steps forward! Thank you! just some minor feedback...

**Comment by RemakingEden** [comment]
@commjoen @tghosth I have made some first changes. I think the grep section looks good but I'm worried the Git secrets section gives away too much. I can't think of a better way to do this though. As this is the first challenge perhaps it is not the end of the world.

**Comment by commjoen** [comment]
@RemakingEden i like how it is made easy as it motivates to move ahead. maybe alternatively you can just mention git-secrets without the added rule, and then detail the commands to be used in the challenge1_hint.adoc ?
You only need to update these 2 files (e.g challenge1.adoc and challenge1_hint.adoc). after that you can let everything be rendered by running `./mvnw clean install` in the root of the project.

File: src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge4Test.java (hunk 2 above)
Comment (RemakingEden) [reply src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge4Test.java]: Yeah sorry was some git shenanigans. I created the MR too early

Comment (commjoen) [reply src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge4Test.java]: no problem sir :). thanks!

File: src/main/resources/explanations/challenge1_hint.adoc [hunk 4]
```diff
@@ -2,10 +2,18 @@ As the text of the challenge says, we are looking for a secret called `password`
...
+- Navigate to the Java code where the check for the right answer happens `cd src/main/java/org/owasp/wrongsecrets/`.
+- Use grep recursively to look for the password string `grep -r password`.
 
-1. Use https://github.com/trufflesecurity/truffleHog[*Trufflehog*] to scan the repository for secrets:
-- Read up on the instructions of the tool at https://github.com/trufflesecurity/truffleHog[Trufflehog] and install it using `pip3 install truffleHog` or use its docker container.
-- Now run Trufflehog at our repository: `trufflehog https://github.com/commjoen/wrongsecrets`. Can you find the value of `public static String password` in the output?
-2. When you do not have the source code available, try to obtain the actual application to reverse engineer it. The application, in this case, is a Jar file, which is a set of java class files together with a few resources.
+2. An automated tool like https://github.com/awslabs/git-secrets[*Git-secrets*] can often help out. In this case it needs a little set up.
+- Clone the repo with `git clone https://github.com/commjoen/wrongsecrets`.
+- Follow the instructions https://github.com/awslabs/git-secrets[here] to install Git leaks.
```
Comment (RemakingEden) [inline src/main/resources/explanations/challenge1_hint.adoc:13]: Ive been using Gitleaks too much recently and its stuck in my head. I'll have to change this to Git Secrets

File: src/main/resources/explanations/challenge1.adoc (hunk 3 above)
Comment (RemakingEden) [reply src/main/resources/explanations/challenge1.adoc]: I think this is done. Let me know what you think of the new layout

**Review by commjoen — APPROVED** [review]
Loving this! Just a final question: which name can I list at the contributor list after merging this? Will `Joss Sparkes`  do?

**Comment by RemakingEden** [comment]
> Loving this! Just a final question: which name can I list at the contributor list after merging this? Will `Joss Sparkes` do?

Glad you like it! Yeah Joss Sparkes is perfect. Thanks Jeroen.

**Pull request merged**